server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessageFastViewProjectionListener.java

```diff
@@ -0,0 +1,105 @@
+import org.apache.james.jmap.draft.model.message.view.MessageFullViewFactory;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.SessionProvider;
+import org.apache.james.mailbox.events.Event;
+import org.apache.james.mailbox.events.Group;
+import org.apache.james.mailbox.events.MailboxListener;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.FetchGroup;
+import org.apache.james.mailbox.model.MessageResult;
+import org.parboiled.common.ImmutableList;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.annotations.VisibleForTesting;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class ComputeMessageFastViewProjectionListener implements MailboxListener.GroupMailboxListener {
+    public static class ComputeMessageFastViewProjectionListenerGroup extends Group {
+
+    }
+
+    static final Group GROUP = new ComputeMessageFastViewProjectionListenerGroup();
+
+    private final MessageIdManager messageIdManager;
+    private final MessageFastViewProjection messageFastViewProjection;
+    private final SessionProvider sessionProvider;
+    private final MessageFullViewFactory messageFullViewFactory;
+
+    @Inject
+    public ComputeMessageFastViewProjectionListener(SessionProvider sessionProvider, MessageIdManager messageIdManager,
+                                                    MessageFastViewProjection messageFastViewProjection,
+                                                    MessageFullViewFactory messageFullViewFactory) {
+        this.sessionProvider = sessionProvider;
+        this.messageIdManager = messageIdManager;
+        this.messageFastViewProjection = messageFastViewProjection;
+        this.messageFullViewFactory = messageFullViewFactory;
+    }
+
+    @Override
+    public Group getDefaultGroup() {
+        return GROUP;
+    }
+
+    @Override
+    public void event(Event event) throws MailboxException {
+        if (event instanceof Added) {
+            MailboxSession session = sessionProvider.createSystemSession(event.getUsername());
+            handleAddedEvent((Added) event, session);
+        }
+    }
+
+    private void handleAddedEvent(Added addedEvent, MailboxSession session) throws MailboxException {
+        Flux.fromIterable(messageIdManager.getMessages(addedEvent.getMessageIds(), FetchGroup.BODY_CONTENT, session))
+            .publishOn(Schedulers.boundedElastic())
+            .flatMap(Throwing.function(messageResult -> Mono.fromCallable(
+                () -> Pair.of(messageResult.getMessageId(), computeFastViewPrecomputedProperties(messageResult)))
+                    .subscribeOn(Schedulers.parallel())))
+            .flatMap(message -> messageFastViewProjection.store(message.getKey(), message.getValue()))
```

Comment:
You need to add `.publishOn(Schedulers.boundedElastic())` before the flatMap otherwise the store operation will execute on the the parallel scheduler. Which it should not as it is an IO bounded operation